### PR TITLE
 check_polkadot.sh: use the branch matching substrate's branch if possible

### PR DESCRIPTION
**Pull request by cecton**

### CODE REVIEW

File: .maintain/gitlab/check_polkadot.sh [hunk 1]
```diff
@@ -73,7 +78,14 @@ then
     git fetch --depth 1 origin refs/pull/${pr_companion}/head:pr/${pr_companion}
     git checkout pr/${pr_companion}
   else
-    boldprint "no companion pr found - building polkadot:master"
+    if git fetch --depth 1 origin \
+      "$CI_MERGE_REQUEST_SOURCE_BRANCH_NAME":branch/"$CI_MERGE_REQUEST_SOURCE_BRANCH_NAME"
```
Comment (gabreal) [inline .maintain/gitlab/check_polkadot.sh]: I think that variable only is available if using gitlab merge requests. see https://docs.gitlab.com/ee/ci/variables/predefined_variables.html#variables-reference
github pull requests are mirrored as branches and their branches are not available by their names.

Comment (cecton) [reply .maintain/gitlab/check_polkadot.sh]: I noticed

**Comment by cecton** [comment]
> I'm not sure if i understand. What's not working properly with the current implementation and what will this improve? Is it tested?

hey o/ This change will allow the companion pr to be found just by its name. If the polkadot branch is exactly the same than the substrate branch name, then it's a match.

**Comment by gabreal** [comment]
I see, unfortunately this won't work that easy as the branch names are not available on gitlab.

**Comment by cecton** [comment]
Yeah it's such a pain! Fortunately it's on the github API. I got it wortking, look :hugs: https://gitlab.parity.io/parity/substrate/-/jobs/440585

**Comment by cecton** [comment]
Well that's wanted by me at least :sweat_smile: I have converted it to a grep command but Perl is actually builtin with Debian and most of the distributions. I also updated the comment

**Comment by cecton** [comment]
@gabreal DIS IS GREEN QUIKLY MERGE PLZ!! :grin:

**Comment by kianenigma** [comment]
would be good to update the contributing docs as well with such changes.

**Pull request merged**